src/org/exist/storage/BrokerPool.java

```diff
@@ -1004,8 +758,9 @@ protected void initialize() throws EXistException, DatabaseConfigurationExceptio
 
                             sync(broker, Sync.MAJOR);
 
+                            //TODO(AR) remove this!
                             //require to allow access by BrokerPool.getInstance();
-                            instances.put(instanceName, this);
+                            //instances.put(instanceName, this);
```

Comment:
Indeed. But this was a bad design decision before as it allowed someone to write code which was accesing a BrokerPool which had not finished being initialized.

Adding it to `instances` at this point is unnecessary, it will be added by `BrokerPools#configure` when initialization is complete. Separating the concerns of BrokerPool and BrokerPools as I have done enabled me to spot such a bug which otherwise was not easy to understand.